viadot/sources/spectrum.py

```diff
@@ -0,0 +1,166 @@
+            name_contains (str, optional): Select by a specific string on table name.
+            search_text (str, optional): Select only tables with the given string in table's properties.
+        """
+        df = wr.catalog.tables(
+            boto3_session=self.session,
+            database=database,
+            name_contains=name_contains,
+            search_text=search_text,
+        )
+
+        return df
+
+    def exists(self, database: str, table: str) -> bool:
+        """
+        Check if a table exists in Glue database.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+
+        Returns:
+            bool: Whether the paths exists.
+        """
+        return wr.s3.does_table_exist(
+            boto3_session=self.session, database=database, table=table
+        )
+
+    def rm(
+        self,
+        database: str,
+        table: str,
+    ):
+        """
+        Deletes table from Glue database.
+
+        Args:
+            path (str): Path to a folder.
+            database (str): AWS Glue  catalog database name.
+            table (str): AWS Glue catalog table name.
+        """
+        table_location = wr.catalog.get_table_location(
+            boto3_session=self.session, database=database, table=table
+        )
+        wr.catalog.delete_table_if_exists(database=database, table=table)
+
+        wr.s3.delete_objects(path=table_location)
+
+    def from_df(
+        self,
+        df: pd.DataFrame,
+        to_path: str,
+        database: str,
+        table: str,
+        extention: str = ".parquet",
+        if_exists: Literal["overwrite", "append"] = "overwrite",
+        partition_cols: List[str] = None,
+        index: bool = False,
+        compression: str = None,
+        sep: str = ",",
+        description: str = "test",
+        columns_comments: Dict[str, str] = None,
+    ):
+        """
+        Upload a pandas `DataFrame` to a csv or parquet file.
+
+        Args:
+            df (pd.DataFrame): Pandas DataFrame
+            to_path (str): Path to a S3 folder. Defaults to None.
+            extention (str): Required file type.
+            database (str): AWS Glue  catalog database name.
+            table (str): AWS Glue catalog table name.
+            partition_cols (List[str]):
+            if_exists (str, optional): 'overwrite' to recreate any possible existing table or 'append' to keep any possible existing table. Defaults to overwrite.
+            index (bool, optional): Write row names (index). Defaults to False.
+            compression (str, optional): Compression style (None, snappy, gzip, zstd).
+            sep (str, optional): Field delimiter for the output file. Defaults to ','.
+            description (str, optional): Glue catalog table description.
+            columns_comments (Dict[str,str], optional) - Glue catalog columns names and the related comments.
```

Comment:
typo: "columns names" -> "column names"

Reply:
Updated as per [🎨 improved structure and updated docstrings](https://github.com/dyvenia/viadot/pull/579/commits/13c71f9ed5e4afc46038950f56bea82ade0fd06c)